### DISCUSSION TITLE
IRS Business rules

### DISCUSSION THREAD

**Discussion by nimish** [ustaxes/UsTaxes#1062] [Q&A]
Does anyone have access to the official biz rules from the IRS? That's how most tax programs are updated.

https://www.irs.gov/pub/irs-pdf/p4164.pdf

**Comment by zakpatterson** [comment]
Accessing this requires being registered as a tax professional with IRS, we don't have anyone like that, I'll try to figure out if we can register: https://www.irs.gov/tax-professionals/instructions-to-access-the-secure-object-repository-sor-mailbox

**Reply by nimish** [reply]
You can reg as a Software Developer. It’s not clear what kind of org you need though and whether you need a tax pro if you aren’t preparing returns for anyone, just making tools.

**Reply by tmm1** [reply]
You can register as a sole proprietorship with your own SSN. No professional status is required, though you may need to send fingerprints into the IRS.

From what I understand, the business rules are just a small validation layer on top of the MeF xml format. There are various ones from earlier years available through https://www.irs.gov/downloads/irs-schema?sort=asc&order=Name&page=0

In the context of UsTaxes, the first step would be to generate and use the MeF format (probably initially with the stylesheets to render forms, and eventually to transmit/e-file) before the business rules could be applied.

**Comment by tmm1** [comment]
Looks like @betson may have some experience with the schemas. I found https://github.com/betson/irs-efile-viewer/tree/master/mef

**Reply by betson** [reply]
Yeah you'd navigate some of what's available here: https://www.irs.gov/e-file-providers/modernized-e-file-mef-schemas-and-business-rules. But the rules have changed since I was using the schemas. They used to be directly accessible from that destination (at least for the 990s), but are now require registration through their portal.

**Reply by tmm1** [reply]
Thanks! It is unclear why they were moved...

If you still have an old copy you can share, that would be helpful. I see the style sheets were checked into your viewer but I didn't see the schemas.

Either way, thanks so much for documenting how this stuff works in your mef/README.md

**Comment by tmm1** [comment]
The business rules I believe are something similar to https://www.irs.gov/pub/irs-utl/f1040brs.pdf

see also https://www.irs.gov/pub/irs-utl/Fixing-Your-Business-Rule-Error.pdf

**Comment by tmm1** [comment]
@divergentdave may also be able to shed some light 

https://github.com/divergentdave/irs-eservices-mailbox-scraper

**Reply by divergentdave** [reply]
It's been a few years since I looked at any of this, but yes, you can sign up as a software developer to get access to the schemas and business rules. There were a number of steps to the process, I think I may have had to get an EIN for myself as a sole proprietorship, as part of it.
